src/paste.rs

```diff
@@ -226,9 +223,6 @@ impl Paste {
             .await
             .map_err(error::ErrorInternalServerError)?
             .to_vec();
-        let config = config
-            .read()
-            .map_err(|_| error::ErrorInternalServerError("cannot acquire config"))?;
```

Comment:
how can you remove retrieving the config when it is needed in line 234?

Reply:
There was a huge amount of duplication for getting the config going on here. By grabbing the read lock right at the top of the function, there's no need to grab it multiple times.

Reply:
ah, the gh diff only showed a few lines so I missed that part. although I think there was a reason for the config to be read at the beginning, e.g. if it was updated on disk and the reload was set to a longer timeframe it would assure the most recent config

Reply:
This is true, but arguably it's best to have a single view of the config for the entire request, otherwise you could get some very weird race conditions.